pkg/trace/agent/agent.go

```diff
@@ -686,13 +686,24 @@ func (a *Agent) runSamplers(now time.Time, ts *info.TagStats, pt traceutil.Proce
 
 func traceContainsError(trace pb.Trace) bool {
 	for _, span := range trace {
-		if span.Error != 0 {
+		if spanIsError(span) {
 			return true
 		}
 	}
 	return false
 }
 
+func spanIsError(span *pb.Span) bool {
+	return span.Error != 0 || spanContainsExceptionSpanEvent(span)
+}
+
+func spanContainsExceptionSpanEvent(span *pb.Span) bool {
+	if hasExceptionSpanEvents, ok := span.Meta["_dd.span_events.has_exception"]; ok && hasExceptionSpanEvents == "true" {
```

Comment:
When will this tag be set?
When:
* _any_ exception is present, or
* When there's [an exception with `exception.escaped` true](https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans/#attributes), or
* A different logic.

I ask because if the logic is consistent, it's best to implement it here, instead of letting every library implement individually.

Reply:
Hi Marco! tag is set for **any** exception (see [otlp.go](https://github.com/DataDog/datadog-agent/blob/2d590bbdf4e30f95675a33db4296e7d70898533c/pkg/trace/api/otlp.go#L520-L525))